<a href="https://colab.research.google.com/github/SteelworkerDB/Proyecto_2/blob/main/Proyecto2_opciones.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
filename='/content/datos.csv'
df_housing=pd.read_csv(filename,encoding='latin-1')
df_housing.head()

Tipo de Propiedad     Sector Total construido      Precio  \
0      Departamento       Same           160 m²  USD160.000   
1              Casa  El Bosque           960 m²  USD800.000   
2              Casa    Cumbayá           275 m²  USD360.000   
3           Terreno      Checa              NaN  USD275.000   
4              Casa    Cumbayá          1740 m²  USD850.000   

                         Referencia  Dormitorios  Baños  Antiguedad  \
0             El Muelle, Esmeraldas          3.0    2.0          15   
1  Sector Unión Nacional, El Bosque          5.0    5.0          38   
2   Cumbayá, junto al Colegio Menor          3.0    3.0  A Estrenar   
3                             Checa          NaN    NaN         NaN   
4                             Tanda          4.0    3.0           8   

  Superficie  Ambientes 24 Hours Security Alcantarillado Parqueadero fijo  \
0        NaN        4.0               NaN            NaN              NaN   
1    1443 m²        6.0               NaN            NaN              NaN   
2      60 m²        4.0               NaN            NaN              NaN   
3    5000 m²        NaN                No             No              NaN   
4        NaN        NaN               NaN            NaN              NaN   

  Seguridad 24Hs  Expensas Fecha de entrega       Fecha  
0            NaN       NaN              NaN  03-04-2022  
1            NaN       NaN              NaN  03-04-2022  
2            NaN       NaN              NaN  03-04-2022  
3            NaN       NaN              NaN  03-04-2022  
4            NaN       NaN              NaN  03-04-2022

In [ ]:
df_housing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 206 entries, 0 to 205
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Tipo de Propiedad  206 non-null    object 
 1   Sector             206 non-null    object 
 2   Total construido   128 non-null    object 
 3   Precio             206 non-null    object 
 4   Referencia         206 non-null    object 
 5   Dormitorios        121 non-null    float64
 6   Baños              126 non-null    float64
 7   Antiguedad         132 non-null    object 
 8   Superficie         137 non-null    object 
 9   Ambientes          108 non-null    float64
 10  24 Hours Security  73 non-null     object 
 11  Alcantarillado     73 non-null     object 
 12  Parqueadero fijo   10 non-null     object 
 13  Seguridad 24Hs     10 non-null     object 
 14  Expensas           7 non-null      float64
 15  Fecha de entrega   4 non-null      object 
 16  Fecha              206 non

**Primera opción: Propiedades en Ecuador**

---

1) Fuente de datos

    Sinthia Guaigua. (2022). DataSet Inmobiliario [Data set]. Zenodo. https://doi.org/10.5281/zenodo.6413272

    Scraped de https://www.inmovision.com.ec/

    Abril 5, 2022

2) Breve descripción de datos

   *Un listado de propiedades en el Ecuador con información detallada que posiblemente no se encuentra detallada en otras bases de datos.*

3) ¿Cuál es el objetivo?

  *Comprender el mercado inmobiliario del Ecuador, lograr predecir costo de una propiedad en base a sus características.*

4) ¿Este es un problema de clasificación o regresión?

*  Regresión.*

5) ¿Cuántas características hay?

*15 (16 contando la fecha de publicación de la información)*

6) ¿Cuántas filas de datos hay?

 *206*

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

  *Demasiados valores perdidos/vacíos.  Características que se observan completas en más de la mitad de las filas son: TIPO DE PROPIEDAD, SECTOR, PRECIO, REFERENCIA, DORMITORIOS, BAÑOS, ANTIGUEDAD, SUPERFICIES, AMBIENTES.  Algunas solo tienen completas un poco más de la mitad del total de filas.*

  *En general, 206 datos no es una gran cantidad de datos, especialmente con tantos valores perdidos.*






---


---



In [3]:
filename='/content/real_state_ecuador_dataset.csv'
df_rentals=pd.read_csv(filename,encoding='latin-1')
df_rentals.head()

Titulo  Precio  Provincia  \
0  Hermoso departamento de arriendo sector iñaquito     500  Pichincha   
1         Hermosa suite de arriendo sector iñaquito     375  Pichincha   
2  Hermoso departamento de oportunidad de arriend...     650  Pichincha   
3  Hermoso departamento de arriendo de 240 metros...     560  Pichincha   
4  Hermoso departamento de arriendo de oportunida...     580  Pichincha   

                                               Lugar Num. dormitorios  \
0               Pichincha, Iñaquito, Quito, Ecuador                2   
1                          Pichincha, Quito, Ecuador                1   
2  Pichincha, Av. República de El Salvador, Quit...                2   
3         Pichincha, Albert Einstein, Quito, Ecuador                3   
4  Pichincha, Urbanización El Condado, Calle C, ...                3   

   Num. banos   Area  Num. garages  
0         2.0   95.0           1.0  
1         1.0   65.0           1.0  
2         2.0  120.0           1.0  
3         3.0  240.0           2.0  
4         4.0  128.0           2.0

In [4]:
df_rentals.shape

(500, 8)

In [5]:
df_rentals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Titulo            500 non-null    object 
 1   Precio            500 non-null    int64  
 2   Provincia         500 non-null    object 
 3   Lugar             500 non-null    object 
 4   Num. dormitorios  360 non-null    object 
 5   Num. banos        403 non-null    float64
 6   Area              482 non-null    float64
 7   Num. garages      348 non-null    float64
dtypes: float64(3), int64(1), object(4)
memory usage: 31.4+ KB


**Segunda Opción: Rentals en Ecuador**

---

1) Fuente de datos

    Erika Paola Martínez Soria, & Eugenia Bezek. (2022). Ecuador Real State data set (1.0) [Data set]. Zenodo. https://doi.org/10.5281/zenodo.6429586

2) Breve descripción de datos

   *Dataset que contiene información general sobre propiedades que se arriendan en Ecuador.  *

3) ¿Cuál es el objetivo?

  *Determinar valores de arriendo en base a características de las propiedades en Ecuador.*

4) ¿Este es un problema de clasificación o regresión?

*  Regresión.*

5) ¿Cuántas características hay?

*8*

6) ¿Cuántas filas de datos hay?

*500*

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

  *Algunos lugares que se encuentren escritos de forma diferente pueden apuntar al mismo sector, a primera vista parece que no se ha llevado una forma estándar de describir el sector en la columna lugar.   *

---
---


In [8]:
filename='/content/steel_strength.csv'
df_steel=pd.read_csv(filename)
df_steel.head()

formula     c    mn    si     cr  \
0  Fe0.620C0.000953Mn0.000521Si0.00102Cr0.000110N...  0.02  0.05  0.05   0.01   
1  Fe0.623C0.00854Mn0.000104Si0.000203Cr0.147Ni0....  0.18  0.01  0.01  13.44   
2  Fe0.625Mn0.000102Si0.000200Cr0.0936Ni0.129Mo0....  0.00  0.01  0.01   8.67   
3  Fe0.634C0.000478Mn0.000523Si0.00102Cr0.000111N...  0.01  0.05  0.05   0.01   
4  Fe0.636C0.000474Mn0.000518Si0.00101Cr0.000109N...  0.01  0.05  0.05   0.01   

      ni    mo     v     n    nb     co     w    al    ti  yield strength  \
0  19.70  2.95  0.01  0.00  0.01  15.00  0.00  0.15  1.55          2411.5   
1   0.01  3.01  0.46  0.04  0.01  19.46  2.35  0.04  0.00          1123.1   
2  13.45  0.82  0.01  0.00  0.01  13.90  0.00  0.39  0.57          1736.3   
3  17.70  3.95  0.01  0.00  0.01  15.00  0.00  0.13  1.47          2487.3   
4  19.40  1.45  0.01  0.00  0.01  14.90  0.00  0.13  1.55          2249.6   

   tensile strength  elongation  
0            2473.5         7.0  
1            1929.2         8.0  
2            1871.8         NaN  
3            2514.9         9.0  
4            2315.0         8.5

In [9]:
df_steel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 312 entries, 0 to 311
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   formula           312 non-null    object 
 1   c                 312 non-null    float64
 2   mn                312 non-null    float64
 3   si                312 non-null    float64
 4   cr                312 non-null    float64
 5   ni                312 non-null    float64
 6   mo                312 non-null    float64
 7   v                 312 non-null    float64
 8   n                 312 non-null    float64
 9   nb                312 non-null    float64
 10  co                312 non-null    float64
 11  w                 312 non-null    float64
 12  al                312 non-null    float64
 13  ti                312 non-null    float64
 14  yield strength    312 non-null    float64
 15  tensile strength  312 non-null    float64
 16  elongation        303 non-null    float64
dt

**Tercera Opción: Resistencia de Acero**

---

1) Fuente de datos

    Hacking Materials Public Data, https://figshare.com/authors/Hacking_Materials/5771387

2) Breve descripción de datos

   *Dataset que contiene información de aceros: resistencia obtenida de ensayos de tracción, y análisis de composición química. *

3) ¿Cuál es el objetivo?

  *Predecir la resistencia del acero en base a su composición.  Es de interés predecir tres valores, Resistencia Última, Resistencia a la Fluencia, y Elongación, la predicción de uno solo de estos valores no es suficiente.*

4) ¿Este es un problema de clasificación o regresión?

*  Regresión.*

5) ¿Cuántas características hay?

*17*

6) ¿Cuántas filas de datos hay?

*312*

7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

  *El aporte de cada elemento dentro de la composición química a la resistencia del acero no es directo.  En ciertos niveles, algunos elementos se apoyan para entregar un acero de mayor resistencia, en otros niveles algunos elementos pueden presentar desventajas como la fragilidad del acero.  Es posible que la cantidad de datos sea insuficiente para capturar la complejidad del problema *

---
---
